alerts/cluster-monitoring-operator/PrometheusScrapeBodySizeLimitHit.md

```diff
@@ -0,0 +1,114 @@
+configured `body_size_limit`.
+By default, there is no limit on the body size of the scraped targets.
+When set, this limit prevents Prometheus from consuming excessive amounts of memory
+when scraped targets return a response that is deemed too large.
+
+## Impact
+
+Metrics coming from targets responding with a body exceeding the configured
+size limit aren't ingested by Prometheus. The targets will be considered as being
+down and they will have their `up` metric set to 0 which may also trigger the
+`TargetDown` alert.
+
+## Diagnosis
+
+We can find the value of the configured body size limit in the configmap `cluster-monitoring-config`
+in the namespace `openshift-monitoring`.
+
+To check the value in configmap, use this command:
+```bash
+oc get cm -n openshift-monitoring cluster-monitoring-config -o yaml | grep enforcedBodySizeLimit
+```
+
+To find out which targets are exceeding the body size limit, we can refer
+to the Openshift Web UI, go to the Observe > Targets page, and check which
+targets are down.
+
+If we need more information than the UI provides, such as discovered labels
+and scrape pool, we can query the Prometheus API endpoint `/api/v1/targets`.
+
+In the field `data.activeTargets` search for those that have `health` field
+different than `up` and check for the `lastError` field for confirmation:
+
+1. Get the name of the secret containing the token of service account `prometheus-k8s`.
+We use the following command and check for the name  `prometheus-k8s-token-[a-z]+`
+in the line `Tokens`. In the example below, the secret name is `prometheus-k8s-token-nwtrf`.
+
+```bash
+$ oc describe sa prometheus-k8s -n openshift-monitoring
+Name:                prometheus-k8s
+Namespace:           openshift-monitoring
+Labels:              app.kubernetes.io/component=prometheus
+                     app.kubernetes.io/instance=k8s
+                     app.kubernetes.io/name=prometheus
+                     app.kubernetes.io/part-of=openshift-monitoring
+                     app.kubernetes.io/version=2.35.0
+Annotations:         serviceaccounts.openshift.io/oauth-redirectreference.prometheus-k8s:
+                       {"kind":"OAuthRedirectReference","apiVersion":"v1","reference":{"kind":"Route","name":"prometheus-k8s"}}
+Image pull secrets:  prometheus-k8s-dockercfg-6wl6b
+Mountable secrets:   prometheus-k8s-dockercfg-6wl6b
+Tokens:              prometheus-k8s-token-nwtrf
+Events:              <none>
+```
+
+2. Decode the token from the secret.
+
+```bash
+token=$(oc get secret $secret_name_here -n openshift-monitoring -o jsonpath={.data.token} | base64 -d)
+```
+
+3. Get the route URL of the Prometheus API endpoint.
+
+```bash
+host=$(oc get route -n openshift-monitoring prometheus-k8s -o jsonpath={.spec.host})
+```
+
+4. List all the scraped targets with health status different than "up".
+
+```bash
+curl -H "Authorization: Bearer $token" -k https://${host}/api/v1/targets | jq '.data.activeTargets[]|select(.health!="up")'
```

Comment:
it might be good to provide an example of the output

Reply:
an example is inserted at line 39.